build/build.go

```diff
@@ -816,6 +816,11 @@ func BuildWithResultHandler(ctx context.Context, nodes []builder.Node, opt map[s
 						rr.ExporterResponse[k] = string(v)
 					}
 					rr.ExporterResponse["buildx.build.ref"] = buildRef
+					if confutil.IsExperimental() && node.Driver.HistoryAPISupported(ctx) {
+						if err := setRecordProvenance(ctx, c, rr, so.Ref, pw); err != nil {
+							return err
+						}
```

Comment:
This extra request might fail for remote builders in some rare cases like DBC. We could consider a throttle logic in this case and if it doesn't succeed, warn but not fail.